cartographer/cloud/internal/client_server_test.cc

```diff
@@ -636,6 +636,56 @@ TEST_P(ClientServerTestByGridType, LoadStateAndDelete) {
   server_->Shutdown();
 }
 
+TEST_P(ClientServerTestByGridType, LoadUnfrozenStateAndDelete) {
+  if (GetParam() == ::cartographer::mapping::GridType::TSDF) {
+    SetOptionsToTSDF2D();
+  }
+  InitializeRealServer();
+  server_->Start();
+  InitializeStub();
+
+  // Load text proto into in_memory_reader.
+  auto reader =
+      ProtoReaderFromStrings(kSerializationHeaderProtoString,
+                             {
+                                 kPoseGraphProtoString,
+                                 kAllTrajectoryBuilderOptionsProtoString,
+                                 kSubmapProtoString,
+                                 kNodeProtoString,
+                                 kTrajectoryDataProtoString,
+                                 kImuDataProtoString,
+                                 kOdometryDataProtoString,
+                                 // TODO: not implemented by 2D pose graph.
```

Comment:
Not sure about this one. Loading unfrozen state with fixed frame pose data leads to a check fail "not implemented":
```
E0809 11:43:14.039064  3494 pose_graph_2d.cc:694] not implemented
F0809 11:43:14.039091  3494 pose_graph_2d.cc:216] Not yet implemented for 2D.
```

Reply:
Fixed frame pose is not yet implemented in 2D. So we would never produce a proto in 2D that constrains that data. Same thing with the kTrajectoryDataProtoString: no proto produced by the 2D pipeline would ever contain such an item. I would therefore just leave these out in the test.

Reply:
Ok. They are however also used in the other LoadStateAndDelete test and there it doesn't crash.

Feel free to remove it if you want to merge, otherwise I'll do it in 2 weeks